### ISSUE TITLE
cf run-task app "ls -al" didn't display output

### ISSUE THREAD

**Issue by drnic** [cloudfoundry/cli#1034]
```
$ cf run-task ghost "ls -al"
Creating task for app ghost in org dingotiles / space blog-production as drnic@starkandwayne.com...
OK

Task 0 has been submitted successfully for execution.
```

I was expecting that the command would be run inside an app container and the stdout/stderr would be displayed to my terminal.

* CLI Version  cf version 6.23.0+c7866be18-2016-12-22
* CC API Endpoint Version  https://api.run.pivotal.io
* Platform & Shell Details  Mac OS X 10.11.6 iTerm

**Comment by drnic** [comment]
I've discovered that the stdout/stderr go to logs. I can understand this as a side effect; but as a user it isn't my expected behavior. I'd like the output returned - I can then process it, filter it, etc in bash script.

**Comment by dkoper** [comment]
We initially wanted to display the logs, or even briefly just the initial logs (especially useful for tasks that fail early because e.g. you have a typo in your command) but this plan changed for a few reasons:
- There currently isn't a way for the CLI to obtain the logs for a particular task:
  The log stream includes logs for the app and all its tasks, keyed by task name, which is not unique.
  Several backend teams need to be involved to solve that problem, but I think they're waiting for more user feedback about the task feature itself to determine the direction to take it to. @SocalNick @ematpl please correct me if I'm wrong.
- In the user testing we did, users were fine to use `cf logs` or even would be using tools like papertrail anyway to view/analyse logs.

There are other concerns as well that we'd need to work out. I believe (long-running/hanged) tasks time out by default after 48 hours. A synchronously logging `cf run-task` command could therefore hang a script for that long. Currently only interactive commands and `cf logs` do not come back as it tails logs and I'm not sure people would be expecting this from other commands.

We are currently discussing the lack of documentation around logging of tasks (all we have is the hint of `logs` in the `SEE ALSO`). Hope to get that addressed soon.

Thanks for the feedback!

**Comment by dkoper** [comment]
Documentation has been improved to mention about `cf logs`:
http://docs.cloudfoundry.org/devguide/using-tasks.html#run-tasks

**Comment by drnic** [comment]
Thanks for reply and bonus docs.
What is the mechanism by which cf push can show logs? Can we borrow that?






On Fri, Dec 23, 2016 at 10:17 AM +1000, "Dies Koper" <notifications@github.com> wrote:












Documentation has been improved to mention about cf logs:

http://docs.cloudfoundry.org/devguide/using-tasks.html#run-tasks

**Comment by dkoper** [comment]
I'll need to ask my team to confirm (or you can check the code), but I believe `cf push` uses the same mechanism as `cf logs` to display staging logs, taking advantage of the knowledge that at push time no instances or tasks for the app are running and the staging process is the only process emitting logs for the app.

**Comment by dkoper** [comment]
Hi @drnic 

We've just released cf CLI 6.23., which shows a tip in `cf run-task -h` to use `cf logs` to inspect the logs.
Hopefully no confusion here anymore.

https://github.com/cloudfoundry/cli#downloads

Thanks for the feedback.

Regards,
Dies Koper
CF CLI PM
